evals/runner.py

```
@@ -0,0 +1,354 @@
+            passed=not problems,
+            detail=detail,
+            duration_ms=result.duration_ms,
+            tokens=result.usage.get("total_tokens", 0) or 0,
+            raw={"reply": result.text[:300]},
+        )
+    finally:
+        # Teardown unconditionally — even when the task crashed or
+        # an assertion raised — so seeded KB rows never leak into the
+        # next case.
+        if setup_applied or "teardown" in case:
+            if "teardown" in case:
```

Ruff (0.15.12)

[warning] Use a single `if` statement instead of nested `if` statements

Combine `if` statements using `and`

(SIM102)